rest_framework/parsers.py

```diff
@@ -61,8 +62,8 @@ def parse(self, stream, media_type=None, parser_context=None):
         encoding = parser_context.get('encoding', settings.DEFAULT_CHARSET)
 
         try:
-            data = stream.read().decode(encoding)
-            return json.loads(data)
+            decoded_stream = codecs.decode(stream, encoding)
```

Comment:
shouldn't this be
```py
decoded_stream = codecs.getreader(encoding)(stream)
```
?

I don't think codes.decode deals with file like objects